zerver/tests/test_retention.py

```diff
@@ -26,68 +26,56 @@ class TestRetentionLib(ZulipTestCase):
     """
 
     def setUp(self) -> None:
-        super().setUp()
-        self.zulip_realm = self._set_realm_message_retention_value('zulip', ZULIP_REALM_DAYS)
-        self.mit_realm = self._set_realm_message_retention_value('zephyr', MIT_REALM_DAYS)
+        self.zulip_realm = get_realm('zulip')
+        self.mit_realm = get_realm('zephyr')
+        self._set_realm_message_retention_value(self.zulip_realm, ZULIP_REALM_DAYS)
+        self._set_realm_message_retention_value(self.mit_realm, MIT_REALM_DAYS)
+
+        # Set publication date of all existing messages to "now", so that we have full
+        # control over what's expired and what isn't.
         Message.objects.all().update(pub_date=timezone_now())
 
-    @staticmethod
-    def _set_realm_message_retention_value(realm_str: str, retention_period: int) -> Realm:
-        realm = Realm.objects.get(string_id=realm_str)
+    def _set_realm_message_retention_value(self, realm: Realm, retention_period: Optional[int]) -> None:
         realm.message_retention_days = retention_period
         realm.save()
-        return realm
 
-    @staticmethod
-    def _change_messages_pub_date(msgs_ids: List[int], pub_date: datetime) -> Any:
-        messages = Message.objects.filter(id__in=msgs_ids).order_by('id')
-        messages.update(pub_date=pub_date)
-        return messages
+    def _change_messages_pub_date(self, msgs_ids: List[int], pub_date: datetime) -> None:
+        Message.objects.filter(id__in=msgs_ids).update(pub_date=pub_date)
 
     def _make_mit_messages(self, message_quantity: int, pub_date: datetime) -> Any:
         # send messages from mit.edu realm and change messages pub date
         sender = self.mit_user('espuser')
         recipient = self.mit_user('starnine')
-        msgs_ids = [self.send_personal_message(sender.email, recipient.email,
-                                               sender_realm='zephyr') for i in
-                    range(message_quantity)]
-        mit_messages = self._change_messages_pub_date(msgs_ids, pub_date)
-        return mit_messages
+        msg_ids = [self.send_personal_message(sender.email, recipient.email,
```

Comment:
This is fine (and definitely better than before), but stylistically, we generally avoid abbreviations in the codebase, because they're harder to understand.  So I would personally have just written `message_ids` here if I were editing it.

(but I think `msg_ids` is perfectly clear, so I probably wouldn't change this one, either).